ios/Classes/SharedStorage.swift

```diff
@@ -18,38 +19,84 @@ public enum SharedStorage: Int {
 }
 
 /// Move file at [filePath] to shared storage [destination] with optional [directory]
-public func moveToSharedStorage(filePath: String, destination: SharedStorage, directory: String) -> String? {
-    guard FileManager().fileExists(atPath: filePath)
-    else {
-        os_log("Cannot move to shared storage: file %@ does not exist", log: log, type: .error, filePath)
-        return nil
-    }
-    let fileUrl = NSURL(fileURLWithPath: filePath)
-    guard let directory = try? directoryForSharedStorage(destination: destination, directory: directory) else {
-        os_log("Cannot move to shared storage: no permission for directory %@", log: log, type: .error, directory)
-        return nil
+public func moveToSharedStorage(filePath: String, destination: SharedStorage, directory: String, complete: @escaping (String?) -> Void) {
+    switch destination {
+    case .images:
+        saveImageToPhotos(filePath: filePath, complete: complete)
```

Comment:
There is a lot of code duplication between `saveVideoToPhotos` and `saveImageToPhotos`. I would suggest creating one method `saveToPhotos(filePath: filePath, destination: SharedStorage, complete: complete)` and use the `destination` parameter to reflect the subtle changes between videos and photos, eg `PHAssetChangeRequest.creationRequestForAssetFromVideo` vs `PHAssetChangeRequest.creationRequestForAssetFromImage`

Reply:
Yeah, would you support narrowing the enum here, to SupportedDestinations?  I just didn't want to either:
- pass through an enum where 4/6 cases are not used
- use composition to handle the different retrieval of the change request and the retrieval of the final url

Reply:
I think it can be done with the existing enum `SharedStorage` and just using an if-statement where required, e.g. 
```
let assetChangeRequest = destination == SharedStorage.video ? 
    PHAssetChangeRequest.creationRequestForAssetFromVideo(atFileURL: fileURL) :
    PHAssetChangeRequest.creationRequestForAssetFromImage(atFileURL: fileURL)
// continue as you have with assetChangeRequest
```